docs/embedding.md

```diff
@@ -42,24 +36,58 @@ one-line script
 ```
 
-Thus, if the HTML snippet was added to `https://my-site.example.com/path/my-web-archive-embed.html`
-then the sw.js should be added such that it is at: `https://my-site.example.com/path/replay/sw.js`.
+If the HTML snippet was added to `https://my-site.example.com/path/my-web-archive-embed.html`
+then the `sw.js` should be added such that it is at: `https://my-site.example.com/path/replay/sw.js`.
 
 That's it! Loading `https://my-site.example.com/path/my-web-archive-embed.html` should now load the web archive.
 
-(Be sure to add width and height styles to the `<replay-web-page>` tag as needed to size the embed).
+Be sure to add width and height styles to the `<replay-web-page>` tag as needed to size the embed.
 
-You can replace `s3://webrecorder-builds/warcs/netpreserve-twitter.warc` with any web archive hosted on your site,
-eg.  `https://my-site.example.com/warcs/my-warc-file.warc`.
+You can replace `s3://webrecorder-builds/wacz/netpreserve-twitter.wacz` with any web archive hosted on your site,
+eg.  `https://my-site.example.com/wacz/my-wacz-file.wacz`.
 
 {:  .fs-3 .pad .bg-grey-lt-100}
 If the file is loaded from a different origin, your site must have CORS access to download the web archive.
 <br>See [CORS restrictions](#cors-restrictions) below for more info.
 
+### Self Hosting
+
+Sometimes it can be desirable to *self-host* the user interface, service worker, and WACZ files. This is useful in situations where you want to prevent tracking by CDNs, or to make it easier to host the content in one place without needing to work out the details of Cross Origin Resource Sharing (CORS). Self-hosting is easy if you follow the same steps as above, but instead of loading the JavaScript and WACZ file from external locations you will need to [download](https://www.jsdelivr.com/package/npm/replaywebpage) the `ui.js` and `sw.js` JavaScript files and put them on the same server as the HTML that you are publishing.
+
+For example if you are publishing a page at `https://my-site.example.com/path/my-web-archive-embed.html` you could adjust the &lt;script&gt; element to load the ui from a location on your website:
+
+```html
+<script src="ui.js"></script>
+
+<replay-web-page source="netpreserve-twitter.wacz" url="https://twitter.com/netpreserve"></replay-web-page>
+```
+
+The following URLs would then need to resolve correctly:
+
+* https://my-site.example.com/path/my-web-archive-embed.html
+* https://my-site.example.com/path/netpreserve-twitter.wacz
+* https://my-site.example.com/path/ui.js
+* https://my-site.example.com/path/replay/sw.js
+
+If you want to embed more than one web archive on your site it can be helpful to centralise the location of the frontend and backend JavaScript and potentially WACZ files. The &lt;replay-web-page&gt; component has a `replayBase` attribute that lets you define the location to load the `sw.js` service worker from. For example if you publish your JavaScript files at:
```

Comment:
```suggestion
If you want to embed more than one web archive on your site it can be helpful to centralise the location of the frontend and backend JavaScript and potentially WACZ files. The &lt;replay-web-page&gt; component has a `replayBase` attribute that lets you define the location to load the `sw.js` service worker from. By default, `replayBase` is set to `./replay/` and so the service worker is loaded from `./replay/sw.js`. For example if you publish your JavaScript files at:
```